cmake/templates/pkgConfig.cmake.in

```diff
@@ -95,8 +95,17 @@ if(NOT "@PROJECT_CMAKE_CONFIG_INCLUDE_DIRS@ " STREQUAL " ")
   set(@PROJECT_NAME@_INCLUDE_DIRS "")
   set(_include_dirs "@PROJECT_CMAKE_CONFIG_INCLUDE_DIRS@")
   foreach(idir ${_include_dirs})
-    if(IS_ABSOLUTE ${idir} AND IS_DIRECTORY ${idir})
-      set(include ${idir})
+    if(IS_ABSOLUTE ${idir})
+      get_filename_component(path ${idir} DIRECTORY)
+      get_filename_component(idir ${idir} NAME)
```

Comment:
Please avoid changing the semantic of a variable in the middle. It makes it more difficult to read and understand the logic.